### DISCUSSION TITLE
thumbnail type.

### DISCUSSION THREAD

**Discussion by mr-pewpy** [OrcaSlicer/OrcaSlicer#4821] [Ideas]
I use thumbnails for orca
but i need a thumnail like prusa for my machine.
here is an example

[reddit thumbnail slicer](https://www.reddit.com/r/elegooneptune2/comments/xt32yf/thumbnails_for_prusaslicersuperslicer/)

![oyk7i7a1q8r91.jpg](https://github.com/SoftFever/OrcaSlicer/assets/85412791/b8038786-7f8c-457a-865b-d68554f1dfa2)

Could in the printer profile settings add ootions for 3 types of thumbnail generation with a min and max px output etc .?

if so how can i do so i really need too.
look at orca slicer section

[ADAPT project](https://sites.google.com/jagmail.southalabama.edu/adapt/home)

**Comment by TexZeTech** [comment]
Try this?
<img src="https://github.com/SoftFever/OrcaSlicer/assets/7254540/63e83dda-0142-4552-a77d-400663f4cd6a" width=80% height=80%>


Also what printer are you using?

Also the link you have to reddit doesn't link to reddit please fix.

**Reply by mr-pewpy** [reply]
so thats the 3 types ?

**Reply by TexZeTech** [reply]
I guess I'm still not understanding what your goal here is. 

You stated "but i need a thumnail like prusa for my machine."

What do you mean by this? The build plate in the background? Resolution? File type? The start of the gcode name? 

Or are you asking for the 3d model of the build plate in Orcaslicer?

Also what printer are you using?

I'm sorry but its just extremely vague.

**Reply by mr-pewpy** [reply]
so if u look at the adapt project we have split beds and i want my thumbnail to preview the model and the bed and texture

**Reply by TexZeTech** [reply]
Well I dropped the question in the OrcaSlicer discord under Orca-help, no response as of yet.

I wasn't able to output the model of the print bed for myself only the models on the bed.

At the very least you should be able to get an image of the print on your touch screen and in mainsail.

I would try the `32x32, 64x64, 300x300` and play around with the Format of G-code thumbnails. I use PNG but maybe the screen you use needs QOI, BTT TT or ColPic. You will have to play around with it. 

And I will once again say the following "Also what printer are you using?" This means list your printer make/model, flavor of firmware, current firmware version number and even what control boards / screen you use. 

Don't just throw a link to your project site and expect people to do the work for you, help us help you.

I do like the concept of the project you have though.

**Reply by mr-pewpy** [reply]
well i mainly use klipper.


ender6
btt manta m5p
rpi cm4 2gb 8gb no-wifi
5x tmc2209
btt pitft 43
microswiss ng
wham bam flex plate pex

adapt project
tl;dr 
drop-in replacement split heater bed system

with
pc machine screws
silicone spacer

and
custom
30w heat plates
teflon insultation layer
aluminum frame mount
canbus controller (stm32g0b1) 9 heaters

**Reply by TexZeTech** [reply]
This is what I found digging around.

wrong product.
~~https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware?tab=readme-ov-file#adding-gcode-thumbnails~~

- `70x70` : TFT24 / TFT28
- `95x80` : TFT43 / TFT50
- `95x95` : TFT35
- `160x140` : TFT70

And if PNG doesn't work try BTT TT

More digging lead me back to this https://github.com/SoftFever/OrcaSlicer/issues/1894

Hope that helped.

**Reply by mr-pewpy** [reply]
[BTT PiTFT43 Github](https://github.com/bigtreetech/BIGTREETECH-TouchScreenHardware/tree/master/BTT%20Pi%20TFT43%20V2.0%20Github)

is the screen

klipperscreen is the output

[KlipperScreen Thumbnail documentation](https://klipperscreen.readthedocs.io/en/latest/Thumbnails/)

**Reply by TexZeTech** [reply]
Can you upload the whole klipperscreen.log after you try to upload somthing?

**Comment by Baloo3D** [comment]
Hi everyone,
I have a printer with an MKS Robin Nano board. In Cura, I can see the thumbnails on the screen, and I believe they are sent in 100x100 and 200x200 resolutions. However, I'm having trouble getting them to work in OrcaSlicer, despite trying different configurations in the G-code thumbnails settings (like 100x100/PNG).
I examined a G-code file generated by Cura, and I noticed it starts with these lines:

;simage:0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
M10086 ;0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000

The file contains a total of 100 lines of code starting with M10086, and some of these lines contain values other than zero. This leads me to believe that the thumbnail might be in a raw format. I suspect that the correct configuration should be 100x100/XXX, but I’m unsure what the appropriate XXX should be. I’ve tried numerous options, but none have worked so far. Unfortunately, the Orca Wiki doesn’t provide any information on this, and I’m not sure what options are available. It’s possible that there may not be a compatible option for the MKS screen format.
If anyone can help, I would greatly appreciate it! Thanks in advance!
